### DISCUSSION TITLE
Bug: micropython-lib/python-ecosys /aiohttp "MQTT over Websocket"

### DISCUSSION THREAD

**Discussion by jomasnash** [micropython/micropython#16329] [Libraries & Drivers]
There is a bug in aiohttp. The problem is if you want to supply a custom header in a ClientSession that connects to a websocket, the header is ignored.

for example: 

```python
headers = {"Sec-WebSocket-Protocol":"mqtt"}
async with aiohttp.ClientSession(headers=headers).ws_connect(url, ssl=sslctx) as ws:
```
Since the first line in "async def handshake(self, uri, ssl, req):"  is "headers = {}", the custom header is lost.

I found this bug when developing a asyncio "MQTT over Websocket" client. I could only connect to the hivemq broker.
All other "free online" brokers (for example "ws://test.mosquitto.org:8080") refused to connect because of missing header entries.

I have created an issue for this, see: https://github.com/micropython/micropython-lib/issues/940

**Comment by bidrohini** [comment]
I hope they will fix it soon.

**Reply by jomasnash** [reply]
See my update in https://github.com/micropython/micropython-lib/issues/940
